src/main/flight/pid.c

```diff
@@ -93,6 +93,21 @@ void pidResetErrorGyro(void)
 
 const angle_index_t rcAliasToAngleIndexMap[] = { AI_ROLL, AI_PITCH };
 
+
+// Low Pass filter for PID controllers
+void pt1_element(int axis, float *delta, uint8_t f_cut, int type) {
+
+	float dT = (float)cycleTime * 0.000001f;
+
+    float    RC;
+    static float    lastDelta[][3] = {{ 0.0f, 0.0f, 0.0f }};
+
```

Comment:
Than I think my gcc compiler might have some optimizer in it

This is output from my looped test code without it. I will add it anyway.

INPUT AXIS=1 DELTA=1000.000000 FCUT=20 TYPE=1
AXIS: 1 OLD_PTERM=0.000000
AXIS: 2 OLD_PTERM=0.000000
AXIS: 3 OLD_PTERM=0.000000
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=0.000000
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000
131.036591


INPUT AXIS=2 DELTA=15.000000 FCUT=20 TYPE=1
AXIS: 1 OLD_PTERM=131.036591
AXIS: 2 OLD_PTERM=0.000000
AXIS: 3 OLD_PTERM=0.000000
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=0.000000
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000
1.965549


INPUT AXIS=1 DELTA=1500.000000 FCUT=20 TYPE=1
AXIS: 1 OLD_PTERM=131.036591
AXIS: 2 OLD_PTERM=1.965549
AXIS: 3 OLD_PTERM=0.000000
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=0.000000
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000
310.420898


INPUT AXIS=3 DELTA=2.000000 FCUT=20 TYPE=0
AXIS: 1 OLD_PTERM=310.420898
AXIS: 2 OLD_PTERM=1.965549
AXIS: 3 OLD_PTERM=0.000000
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=0.000000
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000


INPUT AXIS=3 DELTA=4.000000 FCUT=20 TYPE=0
AXIS: 1 OLD_PTERM=310.420898
AXIS: 2 OLD_PTERM=1.965549
AXIS: 3 OLD_PTERM=0.000000
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=0.262073
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000


INPUT AXIS=3 DELTA=15.000000 FCUT=20 TYPE=1
AXIS: 1 OLD_PTERM=310.420898
AXIS: 2 OLD_PTERM=1.965549
AXIS: 3 OLD_PTERM=0.000000
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=0.751878
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000


INPUT AXIS=1 DELTA=1500.000000 FCUT=20 TYPE=1
AXIS: 1 OLD_PTERM=310.420898
AXIS: 2 OLD_PTERM=1.965549
AXIS: 3 OLD_PTERM=1.965549
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=0.751878
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000
466.299286


INPUT AXIS=3 DELTA=100.000000 FCUT=20 TYPE=1
AXIS: 1 OLD_PTERM=466.299286
AXIS: 2 OLD_PTERM=1.965549
AXIS: 3 OLD_PTERM=1.965549
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=0.751878
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000


INPUT AXIS=1 DELTA=1500.000000 FCUT=20 TYPE=1
AXIS: 1 OLD_PTERM=466.299286
AXIS: 2 OLD_PTERM=1.965549
AXIS: 3 OLD_PTERM=14.811649
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=0.751878
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000


INPUT AXIS=3 DELTA=10.000000 FCUT=20 TYPE=0
AXIS: 1 OLD_PTERM=601.751892
AXIS: 2 OLD_PTERM=1.965549
AXIS: 3 OLD_PTERM=14.811649
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=0.751878
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000


INPUT AXIS=3 DELTA=1.000000 FCUT=20 TYPE=0
AXIS: 1 OLD_PTERM=601.751892
AXIS: 2 OLD_PTERM=1.965549
AXIS: 3 OLD_PTERM=14.811649
AXIS: 1 OLD_DTERM=0.000000
AXIS: 2 OLD_DTERM=0.000000
AXIS: 3 OLD_DTERM=1.963721
AXIS: 1 OLD_GYRO=0.000000
AXIS: 2 OLD_GYRO=0.000000
AXIS: 3 OLD_GYRO=0.000000